controllers/drplacementcontrol.go

```diff
@@ -986,16 +985,24 @@ func (d *DRPCInstance) generateVRGSpecSync() rmn.VRGSyncSpec {
 	}
 }
 
-func dRPolicySupportsRegional(drpolicy *rmn.DRPolicy) bool {
-	return rmnutil.DrpolicyRegionNamesAsASet(drpolicy).Len() > 1
+func dRPolicySupportsRegional(drpolicy *rmn.DRPolicy, drClusters []rmn.DRCluster) bool {
+	return rmnutil.DrpolicyRegionNamesAsASet(drpolicy, drClusters).Len() > 1
 }
 
-func dRPolicySupportsMetro(drpolicy *rmn.DRPolicy) (supportsMetro bool, metroMap map[rmn.Region][]string) {
+func dRPolicySupportsMetro(drpolicy *rmn.DRPolicy, drclusters []rmn.DRCluster) (
+	supportsMetro bool,
+	metroMap map[rmn.Region][]string) {
 	allRegionsMap := make(map[rmn.Region][]string)
 	metroMap = make(map[rmn.Region][]string)
 
-	for _, v := range drpolicy.Spec.DRClusterSet {
-		allRegionsMap[v.Region] = append(allRegionsMap[v.Region], v.Name)
+	for _, managedCluster := range rmnutil.DrpolicyClusterNames(drpolicy) {
+		for _, v := range drclusters {
+			if v.Name == managedCluster {
+				allRegionsMap[v.Spec.Region] = append(
```

Comment:
Is `.Spec.Region` checked anywhere if it is empty given that it is declared as `omitempty`?